pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -71,16 +81,119 @@ func (r *ReconcileNMstate) Reconcile(request reconcile.Request) (reconcile.Resul
 	instance := &nmstatev1alpha1.NMstate{}
 	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if apierrors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Owned objects are automatically garbage collected. For additional cle  anup logic use finalizers.
 			// Return and don't requeue
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 
+	err = r.syncKubernetesNMStateCRDs(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.syncKubernetesNMStateRBAC(instance)
```

Comment:
What is our error handling policy?  should we take care to rollback 'cleanup' ? 
Assuming the following use case:
1.  syncKubernetesNMStateCRDs - succeeded
2.  syncKubernetesNMStateRBAC - failed, so we exit with error. 
So next time reconcile function will be called, I assume that :
syncKubernetesNMStateCRDs  might fail because CRD already exists.

Reply:
I suppose rollback is not part of this PR, we have to really think about it, this PR is like a MVP kind of thing.

Reply:
Typically, there is no rollback in operators. We only need to make sure that if we fail in the second step, we retry. And of course, during the retry we should not fail on the fact that stuff from step 1 is already deployed in the cluster.